### PR TITLE
Remove bayes-opt + pass extra kwargs to self.model.fit()

### PR DESCRIPTION
**Pull request by fmazzasc**
Bayesian optimisation removed as it is considered obsolete and less performing than the optuna one (#144)

### CODE REVIEW

File: hipe4ml/model_handler.py [hunk 1]
```diff
@@ -145,13 +144,16 @@ def fit(self, x_train, y_train):
...
         """
         n_classes = len(np.unique(y_train))
         self._n_classes = n_classes
         if self.training_columns is None:
             self.training_columns = list(x_train.columns)
 
-        self.model.fit(x_train[self.training_columns], y_train)
+        self.model.fit(x_train[self.training_columns], y_train, **kwargs)
 
     def predict(self, x_test, output_margin=True):
```
Comment (mpuccio) [inline hipe4ml/model_handler.py:158]: Shall we have the `**kwargs` also here?

Comment (mpuccio) [reply hipe4ml/model_handler.py:158]: Actually, it might get complicated with `train_test`... feel free to either think about it or ignore this.

**Review by mpuccio — APPROVED** [review]

File: hipe4ml/model_handler.py (hunk 1 above)
Comment (fmazzasc) [reply hipe4ml/model_handler.py]: Thanks @mpuccio, done! But I would still keep train_test as it is

File: tutorials/hipe4ml_tutorial_binary.ipynb [hunk 2]
```diff
@@ -280,17 +280,16 @@
     "id": "fsu0CS_c_N1V"
    },
    "source": [
-    "### **Bayesian optimisation of hyperparameters**\n",
-    "The optimisation of the hyperparameters is a key step to obtain the best performace from the algorithm. In _hipe4ml_ a Bayesian optimisation is used through the method _ModelHandler.optimize\\_params\\_bayes_.\n",
+    "### **Optimisation of hyperparameters with Optuna**\n",
+    "The optimisation of the hyperparameters is a key step to obtain the best performace from the algorithm. In _hipe4ml_ the Optuna package is employed for the optimisation is through the method _ModelHandler.optimize\\_params\\_optuna_. (<https://github.com/optuna/optuna>)\n",
```
Comment (fcatalan92) [inline tutorials/hipe4ml_tutorial_binary.ipynb:284]:
```suggestion
    "The optimisation of the hyperparameters is a key step to obtain the best performace from the algorithm. In _hipe4ml_ the Optuna package is employed for the optimisation through the method _ModelHandler.optimize\\_params\\_optuna_. (<https://github.com/optuna/optuna>)\n",
```

File: examples/optimize_hyperparams_optuna.py [hunk 3]
```diff
@@ -30,7 +31,7 @@ def moving_average(array, n_mov=4):
 if not os.path.exists('downloads/prompt.root'):
     os.system('curl -L https://cernbox.cern.ch/index.php/s/98tSXndX0VhUMbi/download --output downloads/prompt.root')
 
-MOV_AVER_N = 4
+N_JOBS = 40  # set number of jobs to be executed in parallel
```
Comment (fcatalan92) [inline examples/optimize_hyperparams_optuna.py:34]:
```suggestion
N_JOBS = 4  # set number of jobs to be executed in parallel
```
I would put a number sensible for a laptop

File: examples/training_and_testing_example.py [hunk 4]
```diff
@@ -27,17 +27,6 @@
...
-# hyperparams optimization
-HYP_RANGES = {
-    # # defines the maximum depth of a single tree (regularization)
-    'max_depth': (5, 15),
-    # 'learning_rate': (0.01, 0.3),  # learning rate
-    'n_estimators': (5, 10),  # number of boosting trees
-}
-MODEL.optimize_params_bayes(DATA, HYP_RANGES, 'roc_auc')
-
-# train and test the model with the updated hyperparameters
```
Comment (fcatalan92) [inline examples/training_and_testing_example.py:40]: I will keep this test for the new optuna function

**Pull request merged**